src/audio/host.c

```diff
@@ -107,8 +107,8 @@ static void host_update_position(struct comp_dev *dev, uint32_t bytes)
 	if (hd->local_pos >= hd->host_size)
 		hd->local_pos = 0;
 
-	/* NO_IRQ mode if host_period_size == 0 */
-	if (dev->params.host_period_bytes != 0) {
+	/* NO_IRQ mode if period_irq == 0 */
+	if (dev->params.period_irq != 0) {
```

Comment:
you will break skl- platforms, for those platforms, we need period_irq = 1, but you haven't set it anywhere.

I suggest to use no_period_irq(by default with value 0, suitable for skl-, and set it to 1 here for skl+ platforms).